ext/ITensorsHDF5Ext/itensor.jl

```diff
@@ -0,0 +1,36 @@
+using ITensors: IndexSet, inds, itensor, ITensor, storage
+using NDTensors:
+  NDTensors, BlockSparse, Combiner, Dense, Diag, DiagBlockSparse, EmptyStorage
+
+function HDF5.write(parent::Union{HDF5.File,HDF5.Group}, name::AbstractString, T::ITensor)
+  g = create_group(parent, name)
+  attributes(g)["type"] = "ITensor"
+  attributes(g)["version"] = 1
+  write(g, "inds", inds(T))
+  return write(g, "storage", storage(T))
+end
+
+function HDF5.read(
+  parent::Union{HDF5.File,HDF5.Group}, name::AbstractString, ::Type{ITensor}
+)
+  g = open_group(parent, name)
+  if read(attributes(g)["type"]) != "ITensor"
+    error("HDF5 group or file does not contain ITensor data")
+  end
+  # TODO: use Vector{Index} here?
+  inds = read(g, "inds", IndexSet)
```

Comment:
Yeah, we should use `Vector{<:Index}` here, `IndexSet` is meant to be deprecated.

Reply:
Fixed